xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1402,6 +1402,52 @@ namespace VIDEO
       if (!strTrailer.empty())
         movieDetails.m_strTrailer = strTrailer;
 
+      // Deal with 'Disc' subdirectories and add to/create a set
+      if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+              CSettings::SETTING_MYVIDEOS_HANDLEDISCN))
+      {
+        const size_t discPos = pItem->GetDiscPath(movieDetails.m_strFileNameAndPath);
+        if (discPos != 0)
+        {
+          if (movieDetails.m_set.title.empty())
+          {
+            // Get existing sets
+            CFileItemList listItems;
+            const std::string baseDir = "videodb://movies/sets/?ignoreSingleMovieSets=false";
+            if (CDirectory::GetDirectory(baseDir, listItems, "", DIR_FLAG_DEFAULTS))
+            {
+              // See if set exists
+              for (const auto& item : listItems)
+              {
+                const std::string currentSet{
+                    m_database.GetSetById(item->GetVideoInfoTag()->m_iDbId)};
+                if (currentSet == movieDetails.m_strTitle)
+                {
+                  // Set exists
+                  movieDetails.SetSet(movieDetails.m_strTitle);
+                  break;
+                }
+              }
+              if (movieDetails.m_set.title.empty())
+              {
+                // Create set
+                const int idSet = m_database.AddSet(movieDetails.m_strTitle);
+                m_database.SetArtForItem(idSet, MediaTypeVideoCollection, art);
+                movieDetails.SetSet(movieDetails.m_strTitle);
+              }
+            }
+          }
+          // Add 'Disc n' to title (in local language)
+          const size_t discNumEnd = movieDetails.m_strFileNameAndPath.find("/", discPos + 1);
+          const size_t discNumStart = movieDetails.m_strFileNameAndPath.rfind(" ", discNumEnd);
+          const std::string discNum = movieDetails.m_strFileNameAndPath.substr(discNumStart + 1, discNumEnd - discNumStart - 1);
+          movieDetails.m_strTitle =
+              StringUtils::Format(g_localizeStrings.Get(39199),
+                                  movieDetails.m_strTitle, g_localizeStrings.Get(427),
+                                  discNum);
```

Comment:
No need for the second parameter (`#427`). Let's make things not more complicated than needed. See my other comment.

Reply:
I figured that, as the `#427` translation is already there, this will work for more people in their native language right from the start.

Reply:
> this will work for more people in their native language right from the start.

Why? There is no functional change and it does not help translaters - only makes it harder to understand. Trust me, I'm the maintainer of German language for Koi and know what I'm talking about.

with your approach we could end up with having every single word to translate in one string and compiling all multi-word string at runtime with complicated template strings. We need to draw a line.

Reply:
Fair enough. Changed.